lib/app/utils/validation.rb

```diff
@@ -163,5 +181,9 @@ def self.guess_r4_profile(resource)
 
       candidates.first
     end
+
+    def self.resource_contains_code(resource, code)
```

Comment:
This method doesn't seem as general as its name implies. Maybe `observation_contains_code` until it becomes more general?

Reply:
agreed, fixed.